docs/composability/common_solana_terminology.md

```diff
@@ -0,0 +1,89 @@
+
+## Core concepts of Solana
+
+### Accounts
+
+Since Solana programs are stateless, hence the data on Solana is stored in what are known as "accounts". Each account is uniquely identified by a 32-byte address formatted as an Ed25519 PublicKey. This address acts as the account's unique identifier. Some of the key characteristic of accounts are -
+
+1. Accounts can store up to 10MB of data, which may include executable program code or program state.
+2. Accounts require a rent deposit in SOL, proportional to the amount of data stored. This deposit is fully refundable when the account is closed.
+3. Each account has a program "owner". Only the owning program can modify the account's data or reduce its lamport balance, but anyone can increase the balance.
+
+Every account in Solana has the structure called `AccountInfo` which includes the following fields -
+
+- `data`: A byte array that holds the state of an account. If the account is a program (smart contract), it contains the executable program code. This field is commonly known as "account data".
+- `executable`: A boolean flag indicating whether the account is a program.
+- `lamports`: A numerical representation of the account's balance in lamports, the smallest unit of SOL (1 SOL = 1 billion lamports).
+- `owner`: Indicates the public key (program ID) of the program that owns the account.
+
+Solana account types can be divided into two broad categories - Executable and non-executable programs.
+
+1. **Executable Programs**: Executable programs consist of immutable code that can own and create other accounts to store state. This code is written in a language such as Rust and then compiled into eBPF, a form of bytecode.
+
+   The following are the two types of executable accounts -
+
+   - `Native Programs`: Executable native programs are integral to the Solana ecosystem, handling tasks such as maintaining and operating validator nodes. The most notable example is the **System Program**, responsible for creating new system accounts (commonly referred to as "wallets") and transferring SOL. Other examples of native programs include the **Stake Program**, which manages the staking mechanism, and the BPF Loader, which functions similarly to Ethereum’s EVM.
+   - `Custom Programs`: Executable program accounts are pre-built Solana programs designed to create and store other programs. A prime example is the **Solana Program Library (SPL)**, a collection of programs that facilitate various on-chain activities such as creating, swapping, and lending tokens, generating stake pools, and maintaining an on-chain name service. A notable component of the SPL is the Token Program, which creates and manages tokens.
```

Comment:
@YYYurchenko has a point here, I also had doubts when I was reading this. Because on some places you will see the SPLToken program is a native one, but if you open the official documentation you won't find this confirmed. Referring to [this resource](https://solanacookbook.com/core-concepts/programs.html#native-programs-the-solana-program-library-spl) maybe we should have 3 different bullets for executable accounts? Like:
- Native Programs - System Program, Stake Program, Vote Program, etc.
- SPL Programs - Token program, Associated Token Program
- Custom programs - the ones that Yury mention - We can give example with our own Neon EVM program, Orca, etc.

Reply:
The user's or developer's programs (or smart contracts) will come under `Custom Programs`. We didn't describe everything in details because it is going to be a long topic and since solidity developers will mostly focus on the solidity part and programs on Solana are written in Rust.

We will try to make this page a little more developer friendly in the coming versions with more contexts regarding accounts and programs.